src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1818,34 +1891,111 @@ private static void validateStyleModules(Set<Node> checks, Set<Node> configs,
-                                    + moduleName + "' should have matching " + configName + " url")
-                        .that(configUrl)
-                        .isEqualTo(expectedUrl);
+                            + moduleName + "' should have matching " + configName + " url")
+                            .that(configUrl)
+                            .isEqualTo(expectedUrl);
                 }
                 else if ("test".equals(configName)) {
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have matching " + configName + " url")
+                            + moduleName + "' should have matching " + configName + " url")
                             .that(configUrl)
                             .startsWith("https://github.com/checkstyle/checkstyle/"
                                     + "blob/master/src/it/java/com/" + styleName
                                     + "/checkstyle/test/");
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have matching " + configName + " url")
+                            + moduleName + "' should have matching " + configName + " url")
                             .that(configUrl)
                             .endsWith("/" + moduleName + "Test.java");
 
                     assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                                + moduleName + "' should have a test that exists")
+                            + moduleName + "' should have a test that exists")
                             .that(new File(configUrl.substring(53).replace('/',
-                                            File.separatorChar)).exists())
+                                    File.separatorChar)).exists())
                             .isTrue();
                 }
             }
         }
+    }
 
-        assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' has too many configs")
-                .that(itrConfigs.hasNext())
-                .isFalse();
+    private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
+          Iterator<Node> itrConfigs, Set<String> styleChecks, String styleName, String ruleName) {
+        boolean skip = !itrConfigs.hasNext();
+
+        while (itrChecks.hasNext()) {
+            final Node module = itrChecks.next();
+            final String moduleName = module.getTextContent().trim();
+            final String href = module.getAttributes().getNamedItem("href").getTextContent();
+            // until https://github.com/checkstyle/checkstyle/issues/13132
+            final boolean moduleIsConfig = href.startsWith("config_");
+            final boolean moduleIsCheck = href.startsWith("checks/");
+
+            if (!moduleIsConfig && !moduleIsCheck) {
+                skip = true;
```

Comment:
In master, this type of condition was hit on a link to an issue ( `https://github.com/checkstyle/checkstyle/issues/265` ).

Reply:
I am seeing separation a bit better now, skip will skip all test config links if it is set. This could lead to false results where we completely ignore column despite what is in check column.

If column should be empty, it should be validated as empty. If it shouldn't be empty, it should be validated as not empty.

Reply:
> In master, this type of condition was hit on a link to an issue (
> 
> https://github.com/checkstyle/checkstyle/blob/2c232e33c8edccfd7f7658917f7f38208f5a3f2e/src/xdocs/google_style.xml#L199
> 
> ).

yes it is used for this line. Pls refer https://github.com/checkstyle/checkstyle/pull/14925#issuecomment-2157832607